src/App/Fossa/IAT/Resolve.hs

```diff
@@ -0,0 +1,61 @@
+{-# LANGUAGE RecordWildCards #-}
+
+module App.Fossa.IAT.Resolve (
+  resolveShallowGraph,
+) where
+
+import App.Fossa.Analyze.Project (ProjectResult (..))
+import App.Fossa.FossaAPIV1 qualified as Fossa
+import App.Fossa.IAT.Types
+import Control.Algebra
+import Control.Effect.Diagnostics
+import Control.Effect.Lift
+import Data.Maybe
+import DepTypes
+import Fossa.API.Types
+import Graphing qualified
+import Path
+import Srclib.Converter qualified as Srclib
+import Srclib.Types
+import Types
+
+resolveShallowGraph :: (Has (Lift IO) sig m, Has Diagnostics sig m) => Path Abs Dir -> ApiOpts -> [Dependency] -> m SourceUnit
+resolveShallowGraph dir apiOpts shallow = do
+  let (userDefined, direct) = extractUserDefinedBinaries shallow
+  resolvedUserDefined <- resolveUserDefined apiOpts userDefined
+
+  let project = ProjectResult "vsi" dir (Graphing.fromList direct) Complete
+  pure $ toSourceUnit project resolvedUserDefined
+
+toSourceUnit :: ProjectResult -> Maybe [SourceUserDefDep] -> SourceUnit
+toSourceUnit project deps = do
+  let unit = Srclib.toSourceUnit project
+  unit{additionalData = fmap toDepData deps}
+  where
+    toDepData d = AdditionalDepData (Just d) Nothing
+
+resolveUserDefined :: (Has (Lift IO) sig m, Has Diagnostics sig m) => ApiOpts -> [Dependency] -> m (Maybe [SourceUserDefDep])
+resolveUserDefined _ [] = pure Nothing
```

Comment:
[nit] this case turns out to be redundant; `traverse` on the empty list will return an empty list, which will match the `null assertions` check below
```suggestion
```